waterbutler/providers/box/provider.py

```
@@ -253,6 +253,18 @@ def metadata(self, path, raw=False, folder=False, **kwargs):
         return (yield from self._get_folder_meta(path, raw=raw, folder=folder))
 
     @asyncio.coroutine
+    def web_view_link(self, path, **kwargs):
+        resp = yield from self.make_request(
```

Figshare completely breaks since I'm directly building the URL. I'm looking into it now to see if there is a persistent URL I can build for a fileset, or if I'll just need to raise an exception if the given path isn't a file. 

For Box, I can change ```web_view``` to make the appropriate request, like so:
```    @asyncio.coroutine
    def web_view(self, path, **kwargs):
        resp = yield from self.make_request(
                'PUT',
                self.build_url('files' if path.is_file else 'folders', path.identifier),
                data='{"shared_link": {}}',
                expects=(200, ),
                throws=exceptions.WebViewError,
        )
        data = yield from resp.json()
        return data['shared_link']['url']